google/resource_google_folder_test.go

```diff
@@ -0,0 +1,163 @@
+package google
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"os"
+	"testing"
+
+	resourceManagerV2Beta1 "google.golang.org/api/cloudresourcemanager/v2beta1"
+)
+
+func TestAccGoogleFolder_rename(t *testing.T) {
+	skipIfEnvNotSet(t, "GOOGLE_ORG")
+
+	folderDisplayName := "tf-test-" + acctest.RandString(10)
+	newFolderDisplayName := "tf-test-renamed-" + acctest.RandString(10)
+	org := os.Getenv("GOOGLE_ORG")
+	parent := "organizations/" + org
+	folder := &resourceManagerV2Beta1.Folder{}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGoogleFolderDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccGoogleFolder_basis(folderDisplayName, parent),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGoogleFolderExists("google_folder.folder1", folder),
+					testAccCheckGoogleFolderParent(folder, parent),
+					testAccCheckGoogleFolderDisplayName(folder, folderDisplayName),
+				),
+			},
+			resource.TestStep{
+				Config: testAccGoogleFolder_basis(newFolderDisplayName, parent),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGoogleFolderExists("google_folder.folder1", folder),
+					testAccCheckGoogleFolderParent(folder, parent),
+					testAccCheckGoogleFolderDisplayName(folder, newFolderDisplayName),
+				)},
+		},
+	})
+}
+
+func TestAccGoogleFolder_move(t *testing.T) {
```

Comment:
I got this initially confused with rename (move in hierarchies is often used to rename the element, like in filesystems). Consider adding the suffix `_moveParent` to the test name?

Reply:
renamed to `_moveParent`